### ISSUE TITLE
Can we have angular 5 support?

### ISSUE THREAD

**Issue by harryy2510** [jonsamwell/ngx-http-batcher#2]
Hi,
Since http is deprecated and now it is httpClient, Can we have support for angular 5?

**Comment by jonsamwell** [comment]
@harryy2510 yeah I have been meaning to do this since Angular 5 was release I just haven't had time.  The architecture of the new HttpClient is fairly different to that of the old Http so it'll need some thinking about.  I will endeavor to do it in the next few weeks.

**Comment by Evgen14** [comment]
Hi, when you can expect support for angular 5?

**Comment by jonsamwell** [comment]
@Evgen14 Sorry I have been so busy lately.  The current version (v1.10) works with Angular 5 albeit with the now depricated http module.  I need to investigate how to do this with the new HttpClient as I think it might be best to do it via an interceptor.  Any ideas welcome.

**Comment by harryy2510** [comment]
I've tried porting your batcher to angular6. Do you mind looking at it once and adding a few features?

**Comment by Tiberriver256** [comment]
I would be excited to see that!

**Comment by jonsamwell** [comment]
@harryy2510 @Tiberriver256 I do need to get around to porting to Angular 6.  I have some idea around just using an interceptor now.  @harryy2510 I would be very interested to see your ideas.

**Comment by harryy2510** [comment]
Here's the link.

[Stackblitz](https://stackblitz.com/edit/angular-yqifzm)

**Comment by srikrsna** [comment]
[batcher.zip](https://github.com/jonsamwell/ngx-http-batcher/files/2201890/batcher.zip)

Here's the Batcher Using Angular Interceptors with customisable Options.

**Comment by idrisfl** [comment]
I am looking forward to Angular 5 support for this

**Comment by timbyng** [comment]
@srikrsna @harryy2510 Thanks for providing your Angular 6 solutions. @srikrsna, do you have an example on how the BatcherConfigService can be used?

**Comment by srikrsna** [comment]
@timby You can extend the class and override any and all of the methods to create a custom configuration. Once extended you can provide your extension through DI using Angular's `useClass` semantics.

**Comment by JosePFs** [comment]
Thanks @srikrsna @harryy2510 @timby
I've modified interceptor to work with this symfony bundle:
https://github.com/IdeasoftLabs/http-batch-bundle
that implements this draft https://tools.ietf.org/id/draft-snell-http-batch-00.html

Working in Angular 7, 
making requests array with forkjoin (tested)
[batch.zip](https://github.com/jonsamwell/ngx-http-batcher/files/2648575/batch.zip)

Thanks and regards

**Comment by ytilis** [comment]
I've made a new package based heavily on the work by @srikrsna

You can find it here: https://www.npmjs.com/package/ngx-batch-requests

It's configurable via a provider token in AppModule, and I cleaned up and fixed some issues I found in the previous solutions posted in this thread. 

Let me know if anyone wants any additional configurations, you spot any bugs, etc.
